src/nccl_ofi_scheduler.c

```diff
@@ -22,80 +22,54 @@ static inline size_t sizeof_schedule(int num_rails)
 		+ num_rails * sizeof(nccl_net_ofi_xfer_info_t);
 }
 
-void nccl_net_ofi_set_multiplexing_schedule(size_t size, int num_rails,
+int nccl_net_ofi_set_schedule_by_threshold(nccl_net_ofi_threshold_scheduler_t *scheduler,
+					    size_t size,
+					    int num_rails,
 					    size_t align,
 					    nccl_net_ofi_schedule_t *schedule)
 {
+	int ret = 0;
+
+	/* Number of stripes is atleast 1 for zero-sized messages and at most equal to num of rails */
+	int num_stripes = (int) NCCL_OFI_MAX(1, NCCL_OFI_MIN(NCCL_OFI_DIV_CEIL(size, scheduler->min_stripe_size), num_rails));
+	if (OFI_UNLIKELY(num_rails == 0)) {
+		return -1;
+	}
+
+	assert(num_stripes <= num_rails);
+
+	int curr_rail_id, next_rail_id;
+	nccl_net_ofi_mutex_lock(&scheduler->rr_lock);
+
+	/* Retieve and increment multiplex-round-robin counter; wrap around if required */
+	curr_rail_id = scheduler->rr_counter;
+	next_rail_id = (curr_rail_id + num_stripes) % num_rails;
+	scheduler->rr_counter = next_rail_id;
+
+	nccl_net_ofi_mutex_unlock(&scheduler->rr_lock);
+
 	/* Number of bytes left to assign */
 	size_t left = size;
 	/* Offset into message */
 	size_t offset = 0;
 	/* Maximum size of a stripe */
 	size_t max_stripe_size = 0;
 
-	schedule->num_xfer_infos = 0;
-
-	if (OFI_UNLIKELY(num_rails == 0)) return;
+	schedule->num_xfer_infos = num_stripes;
 
-	max_stripe_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(size, num_rails), align) * align;
+	max_stripe_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(size, num_stripes), align) * align;
```

Comment:
There's no reason to declare max_stripe_size on line 56, then initialize it here.

Reply:
Done